core/entities/Module.php

```diff
@@ -308,7 +308,7 @@ public function saveSetting($setting, $value, $uid = 0, $scope = null)
             return framework\Settings::saveSetting($setting, $value, $this->getName(), $scope, $uid);
         }
 
-        public function deleteSetting($setting, $uid = null, $scope = null)
+        public function deleteSetting($setting, $uid = 0, $scope = null)
```

Comment:
:+1: with this explanation if @zegenie is okay with it. I should add a check into the function and log scenario's where the `$uid` is empty:
```php
if (!is_int($uid)) {
    $uid = 0;
    $logger->warning('Non-integer uid tried to delete a Setting');
}
```